docs/grfn_spec.md

```diff
@@ -210,35 +319,41 @@ JSON attribute-value list, with the following schema definition:
 
     <grfn_spec>[attrval] ::=
         "start": <string>
-        "name" : <string>
-        "dateCreated" : <string>
+        "source" : <string>
+        "date_created" : <string>
+        "identifiers" : list of <identifier_spec>
         "functions" : list of <function_spec>
 
+There will be one GrFN spec file corresponding to each source code file
+analyzed by program analysis.
+
 The \"start\" attribute holds the name of the entry point of the
 (Fortran) source code i.e. the PROGRAM module. In the absence of this
-module, this string will remain empty. The \"name\" attribute is used
-to denote the (Fortran) source code that has been analyzed. The
-\"dateCreated\" attribute is a string representing the date+time that
-the current GrFN was generated (this helps resolve what version of the
-program analysis code (e.g., for2py) was used).
+module, this string will remain empty: \"\". 
+
+The \"source\" attribute is used to uniquely identify the source code file that has been analyzed. The \"source\" is represented the same way as a `<namespace_path>` (as described above) except that the final name (the name of the file itself) _will_ include the file extension.
+
+The \"date\_created\" attribute is a string representing the date+time that the current GrFN was generated (this helps resolve what version of the program analysis code (e.g., for2py) was used).
+
+The \"identifiers\" attributes contains a list of `<identifer_spec>`, as has been defined above in the section on Identifiers.
 
 FUTURE:
 
-- We may need to extend \"name\" value to accommodate multiple source
-  files. 
 - It may also be desirable to add an attribute to represent
   the program analysis code version used to generate the GrFN (as
   presumably the program analysis code could evolve and have different
   properties) \-- although \"dateCreated\" may be sufficient.
 
-A (partial) example instance of a JSON attribute-value list generated
-following the `<grfn_spec>`:
+NOTE: variables are not declared at the top-level `<grfn_spec>`, but will be defined in `<function_spec>`s, described below.
+
+A (partial) example instance of the JSON generated for a `<grfn_spec>` of an analyzed file in the path \'crop\_system/yield/crop\_yield.py\' is:
```

Comment:
Is it the case that there is going to be one entry in the "functions" section for every entry in "identifiers"? If so, then I suggest we use a hash-map style datastructure pointed to by functions with the "identifiers" as the keys and the data that would have been stored in "functions" as the objects.

Reply:
Identifiers are just names, they could be used to denote variables or functions. The identifier that is associated with a variable needs to be declared as such. Such declarations, as needed for use in functions, are therefore associated directly with the functions. Let discuss this if it's not clear (possibly I'm not understanding your concern).